File: backend/app/schemas/transcription.py
from pydantic import BaseModel


class TranscribeRequest(BaseModel):
    media_id: str
    model_size: str = "base"  # tiny, base, small, medium, large-v3
    language: str | None = None  # None = auto-detect


class WordTimestamp(BaseModel):
    word: str
    start: float
    end: float
    probability: float


class TranscriptionSegment(BaseModel):
    id: int
    start: float
    end: float
    text: str
    words: list[WordTimestamp] = []


class TranscriptionResult(BaseModel):
    language: str
    segments: list[TranscriptionSegment]

File: frontend/src/types/transcription.ts
export interface WordTimestamp {
  word: string
  start: number
  end: number
  probability: number
}

export interface TranscriptionSegment {
  id: number
  start: number
  end: number
  text: string
  words: WordTimestamp[]
}

export interface TranscriptionResult {
  language: string
  segments: TranscriptionSegment[]
  job_id?: string
}

export interface TranscribeRequest {
  media_id: string
  model_size: string
  language: string | null
}
